src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanSessionsService.kt

```diff
@@ -49,19 +51,41 @@ class ActionPlanSessionsService(
     return actionPlanSessionRepository.save(session)
   }
 
-  fun updateSession(
+  fun updateSessionAppointment(
     actionPlanId: UUID,
     sessionNumber: Int,
-    appointmentTime: OffsetDateTime?,
-    durationInMinutes: Int?
+    appointmentTime: OffsetDateTime,
+    durationInMinutes: Int,
+    updatedBy: AuthUser,
   ): ActionPlanSession {
-
     val session = getActionPlanSessionOrThrowException(actionPlanId, sessionNumber)
-    communityAPIBookingService.book(session, appointmentTime, durationInMinutes)?.let {
-      session.deliusAppointmentId = it
+    val existingAppointment = session.currentAppointment
+
+    val deliusAppointmentId = communityAPIBookingService.book(
+      session.actionPlan.referral,
+      existingAppointment,
+      appointmentTime,
+      durationInMinutes
+    )
+
+    if (existingAppointment == null) {
```

Comment:
What's the benefit of doing this here rather than creating an empty appointment when the session was created? Not an issue, just curious

Reply:
the main benefit is that it allows the appointment entity to contain non nullable fields for duration and time. so a session can exist in an unscheduled state, but it doesn't really make sense for an appointment exist in that way (since at appoint at it's core is just currently time and duration). 

so conceptually the sessions are just containers for appointments. and to create a session without an appointment is exactly that, a session, with an empty appointment set.